src/pages/index.js

```diff
@@ -3,49 +3,125 @@ import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 import {
-  initialCards,
   options,
   formName,
   formCaption,
   forms,
   editBtn,
   addBtn,
+  penBtn,
 } from "../utils/Constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+
+function renderLoading(isLoading, btn, defaultText) {
+  if (isLoading) {
+    btn.textContent = "Saving...";
+  } else {
+    btn.textContent = defaultText;
+  }
+}
+
+function handleDeleteClick(evt) {
+  const element = evt.currentTarget.closest(".element");
+  const popupConfirmDelete = new PopupWithConfirm(
```

Comment:
Every popup should be created only one time and `setEventListeners` should be called also once, otherwise you add listeners again and again to the same elements. This may cause a **`memory leak`**


You can pass `element` into the `open` method, to be able to update the card to be deleted inside the class

```js

function handleDeleteClick(evt) {
  const element = evt.currentTarget.closest(".element");
  popupConfirmDelete.open(element);
}
```